docusaurus/docs/e2e-test-a-plugin/ci.md

```diff
@@ -0,0 +1,86 @@
+---
+id: ci
+title: CI Workflow
+description: How to run end-to-end tests in CI
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - e2e
+  - end-to-end
+  - ci
+sidebar_position: 30
+---
+
+import BEPluginWorkflowNPM from '@snippets/plugin-e2e-ds-workflow.npm.md';
+import BEPluginWorkflowYarn from '@snippets/plugin-e2e-ds-workflow.yarn.md';
+import BEPluginWorkflowPNPM from '@snippets/plugin-e2e-ds-workflow.pnpm.md';
+import FEPluginNPM from '@snippets/plugin-e2e-fe-plugin-workflow.npm.md';
+import FEPluginYarn from '@snippets/plugin-e2e-fe-plugin-workflow.yarn.md';
+import FEPluginPNPM from '@snippets/plugin-e2e-fe-plugin-workflow.pnpm.md';
+
+## Introduction
+
+We recommend using a CI workflow to run end-to-end tests to continuously check for breakages. The following Github workflows will run end-to-end tests against a range of Grafana versions for every PR in your Github repository.
+
+:::note
+These are generic example based on a frontend and backend plugins. You may need to alter or remove some of the steps in the `playwright-tests` job before using it in your plugin.
+:::
+
+<details>
+  <summary> <h3>Backend plugin workflow</h3> </summary>
+  <CodeSnippets
+snippets={[
+{ component: BEPluginWorkflowNPM, label: 'npm' },
+{ component: BEPluginWorkflowYarn, label: 'yarn' },
+{ component: BEPluginWorkflowPNPM, label: 'pnpm' }
+]}
+groupId="package-manager"
+queryString="current-package-manager"
+/>
+</details>
+
+<details>
+  <summary> <h3>Frontend plugin workflow</h3> </summary>
+  <CodeSnippets
+snippets={[
+{ component: BEPluginWorkflowNPM, label: 'npm' },
+{ component: BEPluginWorkflowYarn, label: 'yarn' },
+{ component: BEPluginWorkflowPNPM, label: 'pnpm' }
+]}
+groupId="package-manager"
+queryString="current-package-manager"
+/>
+</details>
+
+### The e2e-versions Action
+
+The example workflows have a job called `Resolve Grafana images` that uses the [e2e-version](https://github.com/grafana/plugin-actions/tree/main/e2e-version) Action to resolve a list of Grafana images. For every image returned, a new job will be fired up that builds the plugin, starts Grafana and runs the end-to-end tests.
+
+The Action supports two modes - `plugin-grafana-dependency` and `version-support-policy`.
+
+#### plugin-grafana-dependency mode
+
+By default, the e2e-version Action will return the most recent grafana-dev image plus all the latest patch releases of every minor version of Grafana Enterprise since the version that was specified as `grafanaDependency` in the [plugin.json](../metadata.md). To avoid starting too many jobs, to output will be capped 6 versions.
```

Comment:
Could we add some "smartness" here return versions depending on how the `grafanaDependency` is specified.

If it is specified like `grafanaDependency: ^9.8.1` we only return the latest patch version of the `9.8` release.

If it is specified `grafanaDependency: >=9.8.1` it would be as you described above etc.

Reply:
That would be nice! Think in general, this Action needs to have more flexibility. For example when we run the tests in CI for the plugin-e2e repo, it uses the `version-support-policy`. That means we're only testing from 9.5, although we have committed to support every version from Grafana 8.5. So would be nice if the Action had mode that allowed you to specify a min version, and also skip the capping of versions. 

I'm making a note on your proposal. 👍